ENH: Launcher

hhslepicka opened the pull request:
This PR adds a luncher to LUCID and a demo folder.

- For now the launcher takes a mandatory argument `beamline` which will be used to query Happi for the devices to be added to the home screen.
- In order to allow for quick testing one can pass `beamline` as `DEMO` and it will launch LUCID with  a random number of fake devices.
- The demo folder contains an example of a `toolbar file` (which will be implemented in a following PR) as well as a Happi sample database and configuration files. That is useful if we want to test the whole scenario in which we load from Happi and generate the UI.

Closes #7

hhslepicka commented:
Kicking Travis now that we have PyQtAds at conda-forge

klauer commented on lucid/launcher.py:
```diff
@@ -0,0 +1,116 @@
+        default='INFO'
+    )
+
+    lucid_args = parser.parse_args()
```
I really like the pattern of making the arg parsing separate from the actual functionality, such that you can programmatically use the launcher without having to make up your own list of string arguments.
e.g.,
```
    args = top_parser.parse_args()
    kwargs = vars(args)
    launcher(**kwargs)
```

where `kwargs` are `{'version', 'beamline', 'toolbar', ...}`. Though you might consider popping `version` out first and handling it specially.

hhslepicka replied:
I like this idea! It sounds really good! The version I think will be handled by the parse_args special version case, right?

klauer replied:
I just learned about that option now, thanks!

klauer commented on demo/launch.sh:
```diff
@@ -0,0 +1,8 @@
+
+CWD=`pwd`
+CONFIG=${CWD}/happi_demo.cfg
+TOOLBAR=${CWD}/demo_toolbar.yml
```
Found this while trying to run the example:
```suggestion
TOOLBAR=${CWD}/example_toolbar.yml
```

hhslepicka replied:
Good catch...

hhslepicka commented:
I think we can iterate at this one until we think it is good... for sure it is a first stab at the problem but if we could at least make it a nice first stab it would be good! ;) So I am fine with keep working and addressing the comments in this PR for now...

hhslepicka commented:
BTW, what an honor of receiving a `Hooray` from the ghost of @teddyrendahl 😆

klauer commented:
After some discussion, I pushed my changes. We'll have to merge fixes from #6 on top of this.

hhslepicka commented:
@klauer says: `Go for it.`
